File: BotBuilderNodeJs/libraries/botframework-schema/src/sharepoint/selectMediaActionParameters.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

/**
 * Sharepoint SelectMediaActionParameters object for select media action
 */
export class SelectMediaActionParameters {
    private mediaType: SelectMediaActionParameters.MediaTypeOption;
    private allowMultipleCapture: boolean;
    private maxSizePerFile?: number;
    private supportedFileFormats?: string[];
    
    /**
     * Initializes a new instance of the SelectMediaActionParameters class
     */
    public SelectMediaActionParameters() {
        // Do nothing
    }

    /**
     * Sets the media type of type string
     */
    public set MediaType(mediaType: SelectMediaActionParameters.MediaTypeOption) {
        this.mediaType = mediaType;
    }

    /**
     * Gets the media type of type string
     */
    public get MediaType(): SelectMediaActionParameters.MediaTypeOption {
        return this.mediaType;
    }

    /**
     * Sets if multiple files can be selected of type boolean
     */
    public set AllowMultipleCapture(allowMultipleCapture: boolean) {
        this.allowMultipleCapture = allowMultipleCapture;
    }

    /**
     * Gets if multiple files can be selected of type boolean
     */
    public get AllowMultipleCapture(): boolean {
        return this.allowMultipleCapture;
    }

    /**
     * Sets the max size per file of type number
     */
    public set MaxSizePerFile(maxSizePerFile: number) {
        this.maxSizePerFile = maxSizePerFile;
    }

    /**
     * Gets the max size per file  of type number
     */
    public get MaxSizePerFile(): number {
        return this.maxSizePerFile;
    }

    /**
     * Sets the supported file formats of type string[]
     */
    public set SupportedFileFormats(supportedFileFormats: string[]) {
        this.supportedFileFormats = supportedFileFormats;
    }

    /**
     * Gets the supported file formats of type string[]
     */
    public get SupportedFileFormats(): string[] {
        return this.supportedFileFormats;
    }
}

export namespace SelectMediaActionParameters {
    export enum MediaTypeOption {
        Image = 1,
        Audio = 4,
        Document = 8
    }
}

File: BotBuilder DotNet/libraries/Microsoft.Bot.Schema/SharePoint/SelectMediaActionParameters.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using System.Collections.Generic;
using System.Text;
using Newtonsoft.Json;

namespace Microsoft.Bot.Schema.SharePoint
{
    /// <summary>
    /// SharePoint parameters for a select media action.
    /// </summary>
    public class SelectMediaActionParameters: ICardActionParameters, IOnCardSelectionActionParameters
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="SelectMediaActionParameters"/> class.
        /// </summary>
        public SelectMediaActionParameters()
        {
            // Do nothing
        }

        public enum MediaType
        {
            /// <summary>
            /// Image
            /// </summary>
            Image = 1,

            /// <summary>
            /// Audio
            /// </summary>
            Audio = 4, 

            /// <summary>
            /// Document
            /// </summary>
            Document = 8
        }

        /// <summary>
        /// Gets or Sets type of media to be selected of type <see cref="MediaType"/>.
        /// </summary>
        /// <value>This value is the type of media to be selected.</value>
        [JsonProperty(PropertyName = "mediaType")]
        public MediaType MediaType { get; set; }

        /// <summary>
        /// Gets or Sets the allow multiple capture property of type <see cref="bool"/>.
        /// </summary>
        /// <value>This value indicates whether multiple files can be selected.</value>
        [JsonProperty(PropertyName = "allowMultipleCapture")]
        public bool AllowMultipleCapture { get; set; }

        /// <summary>
        /// Gets or Sets the max size per file selected of type <see cref="int"/>.
        /// </summary>
        /// <value>This value is the max size per file selected.</value>
        [JsonProperty(PropertyName = "maxSizePerFile")]
        public int MaxSizePerFile { get; set; }

        /// <summary>
        /// Gets or Sets the supported file formats of select media action of type <see cref="string"/>.
        /// </summary>
        /// <value>This value is the supported file formats of select media action.</value>
        [JsonProperty(PropertyName = "supportedFileFormats")]
        public IEnumerable<string> SupportedFileFormats { get; set; }
    }
}
